test/index.js

```diff
@@ -1338,3 +1338,31 @@ describe('parse', function() {
   });
 
 });
+
+describe('regex', function() {
+
+  describe('match', function() {
+
+    it('returns a Just containing an array of Justs', function() {
+      eq(S.match(/abcd/, 'abcdefg').toString(), 'Just([Just("abcd")])');
+    });
+
+    it('supports global patterns', function() {
+      eq(S.match(/[a-z]a/g, 'bananas').toString(),
+        'Just([Just("ba"), Just("na"), Just("na")])');
+    });
+
+    it('supports (optional) capturing groups', function() {
+      eq(S.match(/(good)?bye/, 'bye').toString(),
+        'Just([Just("bye"), Nothing()])');
+      eq(S.match(/(good)?bye/, 'goodbye').toString(),
+        'Just([Just("goodbye"), Just("good")])');
```

Comment:
Let's switch the order of the `'bye'` and `'goodbye'` tests above. For some reason it feels right to me to test the success case (`'goodbye'`) before the partial failure case (`'bye'`).